src/core/recv_buffer.h

```diff
@@ -12,33 +12,56 @@ extern "C" {
 typedef enum QUIC_RECV_BUF_MODE {
     QUIC_RECV_BUF_MODE_SINGLE,      // Only one receive with a single contiguous buffer at a time.
     QUIC_RECV_BUF_MODE_CIRCULAR,    // Only one receive that may indicate two contiguous buffers at a time.
-    QUIC_RECV_BUF_MODE_MULTIPLE     // Multiple independent receives that may indicate up to two contiguous buffers at a time.
+    QUIC_RECV_BUF_MODE_MULTIPLE,    // Multiple independent receives that may indicate up to two contiguous buffers at a time.
+    QUIC_RECV_BUF_MODE_APP_OWNED    // Uses memory buffers provided by the app. Only one receive at a time,
+                                    //   that may indicate up to the number of provided buffers.
 } QUIC_RECV_BUF_MODE;
 
 //
 // Represents a single contiguous range of bytes.
 //
 typedef struct QUIC_RECV_CHUNK {
-    CXPLAT_LIST_ENTRY Link;         // Link in the list of chunks.
-    uint32_t AllocLength : 31;      // Allocation size of Buffer
-    uint32_t ExternalReference : 1; // Indicates the buffer is being used externally.
-    uint8_t Buffer[0];
+    CXPLAT_LIST_ENTRY Link;          // Link in the list of chunks.
+    uint32_t AllocLength       : 31; // Allocation size of Buffer
+    uint32_t ExternalReference : 1;  // Indicates the buffer is being used externally.
+    uint8_t AppOwnedBuffer     : 1;  // Indicates the buffer is managed by the app.
```

Comment:
```suggestion
    uint32_t AllocLength;            // Allocation size of Buffer
    uint8_t ExternalReference  : 1;  // Indicates the buffer is being used externally.
    uint8_t AppOwnedBuffer     : 1;  // Indicates the buffer is managed by the app.
```
Let's leave the 32-bits all for the length, and then put the bits together.